native/src/widget/icon.rs

```diff
@@ -0,0 +1,100 @@
+//! Display an icon.
+use crate::{
+    layout, Element, Hasher, Layout, Length, Point, Rectangle, Widget,
+};
+
+use std::{
+    hash::Hash,
+    path::{Path, PathBuf},
+};
+
+/// A simple icon_loader widget.
+#[derive(Debug, Clone)]
+pub struct Icon {
+    path: PathBuf,
+    size: Length,
+}
```

Comment:
I would name this widget `Svg` and let it be configurable both in width and height.

Reply:
After thinking a bit about it, it may make sense to merge this with `Image` as we had it before. We will probably want to keep the SVG aspect ratio after all.

We can check the extension as you were doing and have an additional `Memory` variant in `image` to deal with caching aware of DPI and viewport dimensions.

Reply:
As I said, that widget can handle svg as well as png or other files since some icons still come in png format. So I think naming it `Svg` would be rather misleading about its capabilities.
I also thought that since icons are usually squares, giving it width and height would be unnecessary and a single size attribute simpler.

Reply:
I was thinking about the image widget too. I would like to give the user more control about aspect ratio of the widget. Maybe an enum with `KeepAspectRatio`, `Stretch` and `KeepAspectRatioCrop` at least. But icons don't need that. 
Also having a separate widget makes integrating themed icons easier in the future.

Reply:
What an icon is depends on each particular application.

To keep things simple, I was proposing that we should instead expose a widget that is able to render only SVG with configurable dimensions. One that can be used to render SVG directly, not just icons.

This way, you could easily create a function `icon` in your application that chooses `Svg` or an `Image` widget based on the path, sets square dimensions, and returns an `Element`.

Reply:
Now the `Icon` widget does all that for you. You don't have to create an `icon` function yourself.
It's very easy to use.

Reply:
Convenient, but not simple.

If our renderer supports SVG, we need a widget that can leverage it to its full extent, not just squares.

Reply:
I could easily merge the svg capability into `Image`. If simplicity is the main goal here, I could try to merge the two pipelines too. That shouldn't be too hard.

Reply:
I am not completely sure about it.

I like the idea of having a dedicated `Svg` widget. It makes everything more explicit and forces users to keep in mind `Svg` and `Image` have different performance implications. It may also help us in the long run to feature-gate some particular widgets and reduce binary size. I'd say let's keep it as different widgets for now.

However, I believe we can reuse the `image` pipeline by adding a new variant to the `Memory` enum to handle the `Svg` primitive. This should reduce the scope of the changes too.

For instance:

```rust
enum Memory {
    Host(HostMemory),
    Device {
        bind_group: Rc<wgpu::BindGroup>,
        width: u32,
        height: u32,
    },
    NotFound,
    Invalid,
}

enum HostMemory {
    Image(image::ImageBuffer<image::Bgra<u8>, Vec<u8>>),
    Svg { /* ... */ },
}
```